internal/repository/step_run.go

```diff
@@ -140,6 +140,8 @@ type StepRunEngineRepository interface {
 
 	GetStepRunForEngine(ctx context.Context, tenantId, stepRunId string) (*dbsqlc.GetStepRunForEngineRow, error)
 
+	ListStepRunsByWorkflowRunId(ctx context.Context, tenantId, workflowRunId string) ([]*dbsqlc.GetStepRunForEngineRow, error)
```

Comment:
This looks like a duplicate of `ListStepRuns` which takes the following options:

```
type ListStepRunsOpts struct {
	JobRunId *string `validate:"omitempty,uuid"`

	WorkflowRunIds []string `validate:"dive,uuid"`

	Status *dbsqlc.StepRunStatus
}
```